app/controllers/events_controller.rb

```diff
@@ -1,61 +1,67 @@
 class EventsController < ApplicationController
-  def index
-    @events = Event.all.order(:event_date)
-  end
+  before_action :authenticate_user!
 
   def new
     @event = Event.new
   end
 
+  def index
+    @events = Event.all.order(:event_date)
+    @play_books = PlayBook.all
+    @players = Player.where(user_id: current_user.id)
+  end
+
   def edit
     @event = Event.find(params[:id])
   end
 
+  def add_player
+    @event = Event.find(params[:id])
+    @player = Player.find(params[:player_id])
+    @event.players << @player
```

Comment:
this should trigger save, so explicit save on line 22 is therefore possibly not required